src/main/java/com/openshift/internal/restclient/model/volume/PersistentVolume.java

```diff
@@ -227,40 +227,30 @@ private IHostPathVolumeProperties createHostPathVolumeProperties() {
      *            the capacity string
      * @param designatedUnit
      *            the designated unit of type {@link MemoryUnit}. Possible values
-     *            'Ki, Mi, Gi, Ti, Pi, Ei'.
+     *            'B, K, Ki, M, Mi, G, Gi, T, Ti, P, Pi, E, Ei'.
      * @return 0 if conversion is not possible, otherwise the numeric representation
      *         converted to designatedUnit. Conversion is not possible when the
      *         designatedUnit cannot contain the unit given by Kubernetes in
      *         integral form.
+     * @throws ArithmeticException
+     *             if computation overflows
      */
     private static long convert(final String capacity, final MemoryUnit designatedUnit) {
         long number = parseCapacityValue(capacity);
         MemoryUnit currentUnit = parseCapacityUnit(capacity);
 
-        int operation = currentUnit.compareTo(designatedUnit);
-        if (operation == 0L) {
-            return number;
-        } else if (operation > 0L) {
-            return safeMultiplicationBy1024(number, operation); // number << (10 * operation);
-        } else {
-            return 0L;
+        long bytes = Math.multiplyExact(number, currentUnit.getFactor());
```

Comment:
Probably should be some log warning message here identifying the value was not convertable